### PR TITLE
docs: change mentions of nsfw to "age restricted"

### PR DESCRIPTION
**Pull request by Soheab**
## Summary

This PR changes almost all mentions of "NSFW" in the docstrings to "age restricted" since discord has done the same for their TOS/Guidelines and in the UI.
 
I'm unsure if we should also change the classes with "NSFW" in the name or at least add them ad aliases if we do.

## Checklist

- [ ] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
  - [ ] I have run `task pyright` and fixed the relevant issues.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: nextcord/channel.py [hunk 1]
```diff
@@ -142,7 +142,7 @@ class TextChannel(abc.Messageable, abc.GuildChannel, Hashable, PinsMixin):
 
         .. note::
 
-            To check if the channel or the guild of that channel are marked as NSFW, consider :meth:`is_nsfw` instead.
+            To check if the channel or the guild of that channel are marked as age restricted, consider :meth:`is_nsfw` instead.
```
Comment (EmreTech) [inline nextcord/channel.py:145]:
```suggestion
            To check if the channel or the guild of that channel are age restricted, consider :meth:`is_nsfw` instead.
```

File: nextcord/channel.py [hunk 2]
```diff
@@ -1548,7 +1548,7 @@ class VoiceChannel(VocalGuildChannel, abc.Messageable):
...
```
Comment (EmreTech) [inline nextcord/channel.py:1551]:
```suggestion
            To check if the channel or the guild of that channel are age restricted, consider :meth:`is_nsfw` instead.
```

File: nextcord/channel.py [hunk 3]
```diff
@@ -2209,11 +2209,11 @@ class CategoryChannel(abc.GuildChannel, Hashable):
         The position in the category list. This is a number that starts at 0. e.g. the
         top category is position 0.
     nsfw: :class:`bool`
-        If the channel is marked as "not safe for work".
+        If the channel is marked as age restricted.
 
         .. note::
 
-            To check if the channel or the guild of that channel are marked as NSFW, consider :meth:`is_nsfw` instead.
+            To check if the channel or the guild of that channel are marked as age restricted, consider :meth:`is_nsfw` instead.
```
Comment (EmreTech) [inline nextcord/channel.py:2216]:
```suggestion
            To check if the channel or the guild of that channel are age restricted, consider :meth:`is_nsfw` instead.
```

File: nextcord/ext/application_checks/core.py [hunk 4]
```diff
@@ -659,7 +659,7 @@ async def predicate(interaction: Interaction) -> bool:
 
 
 def is_nsfw() -> AC:
-    """A :func:`.check` that checks if the channel is a NSFW channel.
+    """A :func:`.check` that checks if the channel is a age restricted channel.
```
Comment (EmreTech) [inline nextcord/ext/application_checks/core.py:662]:
```suggestion
    """A :func:`.check` that checks if the channel is age restricted.
```

File: docs/migrating.rst [hunk 5]
```diff
@@ -916,7 +916,7 @@ Along with this change, a couple of new checks were added.
     - This is actually powered by a different function, :meth:`~ext.commands.Bot.is_owner`.
     - You can set the owner ID yourself by setting :attr:`.Bot.owner_id`.
 
-- :func:`~ext.commands.is_nsfw` checks if the channel the command is in is an NSFW channel.
+- :func:`~ext.commands.is_nsfw` checks if the channel the command is in is an age restricted channel.
```
Comment (EmreTech) [inline docs/migrating.rst:919]:
```suggestion
- :func:`~ext.commands.is_nsfw` checks if the channel the command is in is age restricted.
```

**Review by EmreTech — CHANGES_REQUESTED** [review]
Few grammar mistakes I noticed.

> I'm unsure if we should also change the classes with "NSFW" in the name or at least add them ad aliases if we do.

As for this, we should definitely add aliases for the old class names because otherwise that would be a breaking change and would be left for 3.0.

**Review by Skelmis — APPROVED** [review]

File: docs/api.rst [hunk 6]
```diff
@@ -2851,48 +2851,50 @@ of :class:`enum.Enum`.
...
+            Checks if a Age Restriction level is lower or equal to another.
 
     .. attribute:: default
 
         The guild has not been categorised yet.
 
     .. attribute:: explicit
 
-        The guild contains NSFW content.
+        The guild contains age restriced content.
```
Comment (EmreTech) [inline docs/api.rst:2889]:
```suggestion
        The guild contains age restricted content.
```

File: docs/api.rst [hunk 7]
```diff
@@ -2851,48 +2851,50 @@ of :class:`enum.Enum`.
...
 
     .. attribute:: safe
 
-        The guild does not contain any NSFW content.
+        The guild does not contain any age restriced content.
```
Comment (EmreTech) [inline docs/api.rst:2893]:
```suggestion
        The guild does not contain any age restricted content.
```

File: docs/api.rst [hunk 8]
```diff
@@ -2851,48 +2851,50 @@ of :class:`enum.Enum`.
...
 
     .. attribute:: age_restricted
 
-        The guild may contain NSFW content.
+        The guild may contain age restriced content.
```
Comment (EmreTech) [inline docs/api.rst:2897]:
```suggestion
        The guild may contain age restricted content.
```

File: docs/api.rst [hunk 9]
```diff
@@ -2851,48 +2851,50 @@ of :class:`enum.Enum`.
...
-            Checks if two NSFW levels are equal.
+            Checks if two Age Restriction levels are equal.
         .. describe:: x != y
 
-            Checks if two NSFW levels are not equal.
+            Checks if two Age Restriction levels are not equal.
         .. describe:: x > y
 
-            Checks if a NSFW level is higher than another.
+            Checks if a Age Restriction level is higher than another.
```
Comment (EmreTech) [inline docs/api.rst:2872]:
```suggestion
            Checks if an Age Restriction level is higher than another.
```

File: docs/api.rst [hunk 10]
```diff
@@ -2851,48 +2851,50 @@ of :class:`enum.Enum`.
...
         .. describe:: x < y
 
-            Checks if a NSFW level is lower than another.
+            Checks if a Age Restriction level is lower than another.
```
Comment (EmreTech) [inline docs/api.rst:2875]:
```suggestion
            Checks if an Age Restriction level is lower than another.
```

File: docs/api.rst [hunk 11]
```diff
@@ -2851,48 +2851,50 @@ of :class:`enum.Enum`.
...
         .. describe:: x >= y
 
-            Checks if a NSFW level is higher or equal to another.
+            Checks if a Age Restriction level is higher or equal to another.
```
Comment (EmreTech) [inline docs/api.rst:2878]:
```suggestion
            Checks if an Age Restriction level is higher or equal to another.
```

File: docs/api.rst [hunk 12]
```diff
@@ -2851,48 +2851,50 @@ of :class:`enum.Enum`.
...
         .. describe:: x <= y
 
-            Checks if a NSFW level is lower or equal to another.
+            Checks if a Age Restriction level is lower or equal to another.
```
Comment (EmreTech) [inline docs/api.rst:2881]:
```suggestion
            Checks if an Age Restriction level is lower or equal to another.
```

Comment (EmreTech) [inline docs/api.rst:2881]:
```suggestion
            Checks if an Age Restriction level is lower or equal to another.
```

Comment (EmreTech) [inline docs/api.rst:2881]:
```suggestion
            Checks if an Age Restriction level is lower or equal to another.
```

File: docs/whats_new.rst [hunk 13]
```diff
@@ -1053,7 +1053,7 @@ Bug Fixes
 - Use message creation time for reference time when computing cooldowns.
     - This prevents cooldowns from triggering during e.g. a RESUME session.
 - Fix the default :func:`on_command_error` to work with new-style cogs (:dpyissue:`2094`)
-- DM channels are now recognised as NSFW in :func:`~.commands.is_nsfw` check.
+- DM channels are now recognised as age restricted in :func:`~.commands.is_nsfw` check.
```
Comment (EmreTech) [inline docs/whats_new.rst:1056]: I don't think it's necessary to update an older changelog to use newer terminology.

File: nextcord/channel.py [hunk 14]
```diff
@@ -870,7 +922,14 @@ class ForumChannel(abc.GuildChannel, Hashable):
     position: :class:`int`
         The position in the channel list, where the first channel is ``0``.
     nsfw: :class:`bool`
-        If this channel is marked as NSFW.
+        Alias for :attr:`.age_restricted`.
+
+        .. versionadded:: 2.5
```
Comment (EmreTech) [inline nextcord/channel.py:927]:
```suggestion
        .. versionchanged:: 2.5
```

File: nextcord/ext/commands/errors.py [hunk 15]
```diff
@@ -714,22 +715,30 @@ def __init__(self, missing_roles: List[Union[str, int]]) -> None:
...
+    """Exception raised when a channel is not age restricted.
 
     This inherits from :exc:`CheckFailure`.
 
-    .. versionadded:: 1.1
+    There is an alias for this called ``NSFWChannelRequired``.
+
+    .. versionadded:: 2.5
+    .. versionchanged:: 2.5
+        Renamed from ``NSFWChannelRequired`` to :exc:`AgeRestrictedChannelRequired`. ``NSFWChannelRequired`` is now an alias.
```
Comment (EmreTech) [inline nextcord/ext/commands/errors.py:727]: `versionadded` and `versionchanged` cannot be mixed in the same class.

File: nextcord/ext/commands/errors.py [hunk 16]
```diff
@@ -714,22 +715,30 @@ def __init__(self, missing_roles: List[Union[str, int]]) -> None:
...
 
     Parameters
     ----------
     channel: Union[:class:`.abc.GuildChannel`, :class:`.Thread`]
-        The channel that does not have NSFW enabled.
+        The channel is not age restricted
```
Comment (EmreTech) [inline nextcord/ext/commands/errors.py:732]:
```suggestion
        The channel is not age restricted.
```

Comment (EmreTech) [inline nextcord/ext/commands/errors.py:732]:
```suggestion
        The channel is not age restricted.
```

**Review by EmreTech — COMMENTED** [review]
Mostly spelling and grammar errors.

**Comment by EmreTech** [comment]
@Soheab any updates on this?

**Comment by Soheab** [comment]
> @Soheab any updates on this?

Resolved all the reviews!

File: nextcord/channel.py [hunk 17]
```diff
@@ -879,7 +931,14 @@ class ForumChannel(abc.GuildChannel, Hashable):
     position: :class:`int`
         The position in the channel list, where the first channel is ``0``.
     nsfw: :class:`bool`
-        If this channel is marked as NSFW.
+        Alias for :attr:`.age_restricted`.
+
+        .. versionchanged:: 2.5
+            This is now an alias for :attr:`.age_restricted`.
+    age_restricted: :class:`bool`
+        If this channel is age restricted.
```
Comment (EmreTech) [inline nextcord/channel.py:939]:
```suggestion
        Whether the channel is age restricted.
```

File: nextcord/channel.py [hunk 18]
```diff
@@ -1633,13 +1734,20 @@ class VoiceChannel(VocalGuildChannel, abc.Messageable):
...
     nsfw: :class:`bool`
-        If the channel is marked as "not safe for work".
+        Alias for :attr:`.age_resricted`.
+
+        .. versionadded:: 2.1
+
+        .. versionchanged:: 2.5
+            This is now an alias for :attr:`.age_resricted`.
+    age_resricted: :class:`bool`
+        If the channel is age restricted.
```
Comment (EmreTech) [inline nextcord/channel.py:1744]:
```suggestion
        Whether the channel is age restricted.
```

File: nextcord/channel.py [hunk 19]
```diff
@@ -1633,13 +1734,20 @@ class VoiceChannel(VocalGuildChannel, abc.Messageable):
...
 
         .. note::
 
-            To check if the channel or the guild of that channel are marked as NSFW, consider :meth:`is_nsfw` instead.
+            To check if the channel or the guild of that channel are age restricted, consider :meth:`.is_age_resricted` instead.
```
Comment (EmreTech) [inline nextcord/channel.py:1748]:
```suggestion
            To check whether the channel or the guild of that channel are age restricted, consider :meth:`.is_age_restricted` instead.
```

File: nextcord/channel.py [hunk 20]
```diff
@@ -2346,11 +2466,18 @@ class CategoryChannel(abc.GuildChannel, Hashable):
         The position in the category list. This is a number that starts at 0. e.g. the
         top category is position 0.
     nsfw: :class:`bool`
-        If the channel is marked as "not safe for work".
+        Alias for :attr:`.age_restricted`.
+
+        .. versionchanged:: 2.5
+            This is now an alias for :attr:`.age_restricted`.
+    age_restricted: :class:`bool`
+        If the channel is age restricted.
```
Comment (EmreTech) [inline nextcord/channel.py:2474]:
```suggestion
        Whether the channel is age restricted.
```

File: nextcord/channel.py [hunk 21]
```diff
@@ -2346,11 +2466,18 @@ class CategoryChannel(abc.GuildChannel, Hashable):
...
 
         .. note::
 
-            To check if the channel or the guild of that channel are marked as NSFW, consider :meth:`is_nsfw` instead.
+            To check if the channel or the guild of that channel are age restricted, consider :meth:`is_age_restricted` instead.
```
Comment (EmreTech) [inline nextcord/channel.py:2478]:
```suggestion
            To check whether the channel or the guild of that channel are age restricted, consider :meth:`is_age_restricted` instead.
```

File: nextcord/ext/application_checks/errors.py [hunk 22]
```diff
@@ -231,20 +232,28 @@ class ApplicationNotOwner(ApplicationCheckFailure):
...
 
 
-class ApplicationNSFWChannelRequired(ApplicationCheckFailure):
-    """Exception raised when a channel does not have the required NSFW setting.
+class ApplicationAgeRestrictedChannelRequired(ApplicationCheckFailure):
+    """Exception raised when a channel is not age restricted.
 
     This inherits from :exc:`~.ApplicationCheckFailure`.
 
+    There is an alias for this called ``ApplicationNSFWChannelRequired``
```
Comment (EmreTech) [inline nextcord/ext/application_checks/errors.py:240]:
```suggestion
    There is an alias for this called ``ApplicationNSFWChannelRequired``.
```

File: nextcord/ext/commands/errors.py [hunk 23]
```diff
@@ -714,22 +715,30 @@ def __init__(self, missing_roles: List[Union[str, int]]) -> None:
...
+
+    .. versionadded:: 2.5
+        Renamed from ``NSFWChannelRequired`` to :exc:`AgeRestrictedChannelRequired`. ``NSFWChannelRequired`` is now an alias.
 
     Parameters
     ----------
     channel: Union[:class:`.abc.GuildChannel`, :class:`.Thread`]
-        The channel that does not have NSFW enabled.
+        The channel is not age restricted
+        The channel is not age restricted.
```
Comment (EmreTech) [inline nextcord/ext/commands/errors.py:732]:
```suggestion
        The channel that is not age restricted.
```

**Review by EmreTech — CHANGES_REQUESTED** [review]
Remember to update the all of the `versionadded` & `versionchanged` to be 2.6.

**Review by ooliver1 — COMMENTED** [review]
Not all too sure about this, since the developer documentation uses "nsfw" all over, the field `nsfw` and the descriptions for those fields say "nsfw".

**Comment by Soheab** [comment]
> Not all too sure about this, since the developer documentation uses "nsfw" all over, the field `nsfw` and the descriptions for those fields say "nsfw".

Afaik discord calls it "age restricted" or something similar in the UI, things like their ToS (and promotional material ?).

**Comment by Soheab** [comment]
> > Not all too sure about this, since the developer documentation uses "nsfw" all over, the field `nsfw` and the descriptions for those fields say "nsfw".
> 
> Afaik discord calls it "age restricted" or something similar in the UI, things like their ToS (and promotional material ?).

Like the following on mobile

**Comment by ooliver1** [comment]
The UI also says servers, but we use "guilds" as the terminology to match the Discord Developer docs.

**Comment by ooliver1** [comment]
Closed for now due to the above reason that the API naming is "nsfw" and not "age restricted".

**Pull request closed without merging**